[FEATURE] Add module and moduleData backend request attributes (#2375)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/RequestLifeCycle/RequestAttributes/ModuleData.rst

..  include:: /Includes.rst.txt

..  index::
    Request attribute; ModuleData
..  _typo3-request-attribute-module-data:

==========
ModuleData
==========

..  versionadded:: 12.0

The :php:`moduleData` backend request attribute is available when a backend
module is requested. It holds the object :php:`\TYPO3\CMS\Backend\Module\ModuleData`
which contains the stored module data that might have been overwritten through

> the current request (with GET/POST).

Reviewer: ```suggestion
the current request (with `GET`/`POST`).
```

Through the module registration one can define, which properties can be

> overwritten via GET/POST and their default value.

Reviewer: ```suggestion
overwritten via `GET`/`POST` and their default value.
```

The whole determination is done before the requested route target - usually a
backend controller - is called. This means, the route target can just read the

> final module data and does no longer have to fiddle around with overwriting and

Reviewer: no longer? we should only write about how things used to be when the context is very clear

persisting the data manually.

..  note::

>     It's still possible to store and retrieve arbitrary module data. The

Reviewer: ```suggestion
    It is still possible to store and retrieve arbitrary module data. The
```
